<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Results from the Neale Lab" />


<title>SNP Heritability for Phenotype 2040</title>

<script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="site_libs/plotly-binding-4.8.0.9000/plotly.js"></script>
<script src="site_libs/typedarray-0.1/typedarray.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/plotly-htmlwidgets-css-1.41.3/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="site_libs/plotly-main-1.41.3/plotly-latest.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-grid-3.4.1/bootstrap-grid.min.css" rel="stylesheet" />
<!-- setup favicons -->
<link rel="apple-touch-icon-precomposed" sizes="57x57" href="icon/apple-touch-icon-57x57.png" />
<link rel="apple-touch-icon-precomposed" sizes="114x114" href="icon/apple-touch-icon-114x114.png" />
<link rel="apple-touch-icon-precomposed" sizes="72x72" href="icon/apple-touch-icon-72x72.png" />
<link rel="apple-touch-icon-precomposed" sizes="144x144" href="icon/apple-touch-icon-144x144.png" />
<link rel="apple-touch-icon-precomposed" sizes="60x60" href="icon/apple-touch-icon-60x60.png" />
<link rel="apple-touch-icon-precomposed" sizes="120x120" href="icon/apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon-precomposed" sizes="76x76" href="icon/apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon-precomposed" sizes="152x152" href="icon/apple-touch-icon-152x152.png" />
<link rel="icon" type="image/png" href="icon/favicon-196x196.png" sizes="196x196" />
<link rel="icon" type="image/png" href="icon/favicon-96x96.png" sizes="96x96" />
<link rel="icon" type="image/png" href="icon/favicon-32x32.png" sizes="32x32" />
<link rel="icon" type="image/png" href="icon/favicon-16x16.png" sizes="16x16" />
<link rel="icon" type="image/png" href="icon/favicon-128.png" sizes="128x128" />
<meta name="application-name" content="&nbsp;"/>
<meta name="msapplication-TileColor" content="#FFFFFF" />
<meta name="msapplication-TileImage" content="icon/mstile-144x144.png" />
<meta name="msapplication-square70x70logo" content="icon/mstile-70x70.png" />
<meta name="msapplication-square150x150logo" content="icon/mstile-150x150.png" />
<meta name="msapplication-wide310x150logo" content="icon/mstile-310x150.png" />
<meta name="msapplication-square310x310logo" content="icon/mstile-310x310.png" />


<!-- Basic analytics to track views and downloads -->
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-106608402-1', 'auto');
  ga('set', 'anonymizeIp', true);
  ga('send', 'pageview');

</script>

<script>
/**
* Function that tracks a click on an outbound link in Analytics.
# modifications from: https://hacks.mozilla.org/2016/01/google-analytics-privacy-and-event-tracking/
*/
var trackOutboundLink = function(url) {
    // check if the GA object exists and that it has initialized
    if(typeof ga !== 'undefined' && ga.loaded) {
        // if yes, rely on GA to follow link
        ga('send', 'event', 'outbound', 'click', url, {
            'transport': 'beacon',
            'hitCallback': function(){document.location = url;}
        });
    } else {
        // if not, follow link ourselves
        document.location = url;
    }
}
</script>




<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="sandflat.min.css" type="text/css" />
<link rel="stylesheet" href="dt_style.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">UKB Heritability</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="h2_browser.html">Browser</a>
</li>
<li>
  <a href="downloads.html">Downloads</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Methods
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="details.html">Overview</a>
    </li>
    <li>
      <a href="select_topline.html">Selecting Primary GWAS</a>
    </li>
    <li>
      <a href="confidence.html">Confidence Criteria</a>
    </li>
    <li>
      <a href="significance.html">Significance Thresholds</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Plots
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="viz_h2.html">Heritability and Intercept</a>
    </li>
    <li>
      <a href="viz_annot.html">Partitioned Annotations</a>
    </li>
    <li>
      <a href="viz_sampsize.html">Sample Size Scaling</a>
    </li>
    <li>
      <a href="viz_R1vR2.html">Round 1 vs. Round 2</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Archive
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="round1_details.html">Round 1 Methods</a>
    </li>
    <li>
      <a href="round1_h2_browser.html">Round 1 Browser</a>
    </li>
    <li>
      <a href="round1_h2_univar_browser.html">Round 1 Univariate Browser</a>
    </li>
    <li>
      <a href="round1_plots_home.html">Round 1 Plots</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    About
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="credits.html">Credits</a>
    </li>
    <li>
      <a href="https://github.com/Nealelab/UKBB_ldsc">Site Github</a>
    </li>
    <li>
      <a href="https://github.com/Nealelab/UKBB_ldsc_scripts">Analysis Github</a>
    </li>
    <li>
      <a href="https://www.nealelab.is/uk-biobank">Neale Lab UKB</a>
    </li>
  </ul>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">SNP Heritability for Phenotype 2040</h1>
<h4 class="author"><em>Results from the <a href="credits.html">Neale Lab</a></em></h4>
<h4 class="date"><em>Last updated 2022-10-11</em></h4>

</div>


<p><br></p>
<div class="panel panel-info">
<div class="panel-body">
<h3 id="phenotype-risk-taking"><strong>Phenotype:</strong> Risk taking</h3>
<p>This phenotype can be found on the UK Biobank Showcase for <a href="http://biobank.ctsu.ox.ac.uk/crystal/field.cgi?id=2040">code 2040</a>. Neale Lab GWAS results are available for 348,549 unrelated individuals of European ancestry. This is a binary phenotype with 90,500 cases and 258,049 controls.</p>
</div>
</div>
<p><br></p>
<div class="panel panel-primary">
<div class="panel-heading">
<h3 class="panel-title">
Primary Results
</h3>
</div>
<div class="panel-body">
<h4 id="estimated-snp-heritability-0.0914-se0.00598-p4.54e-53"><strong>Estimated SNP heritability:</strong> 0.0914 (se=0.00598, p=4.54e-53)</h4>
<h4 id="significance-level-strongly-significant-after-multiple-testing-correction"><strong><a href="significance.html">Significance level</a>:</strong> strongly significant after multiple testing correction</h4>
<h4 id="confidence-rating-high"><strong><a href="confidence.html">Confidence rating</a>:</strong> high</h4>
<p>
</p>
<p><em>Note:</em> SNP heritability for this binary outcome is reported on the liability scale, assuming that the population prevalence matches the prevalence in the UK Biobank analysis set (0.26). This may be unreliable if the outcome was limited to a subset of individuals, or if the UK Biobank study population is not representative for this phenotype (<a href="https://academic.oup.com/aje/article/doi/10.1093/aje/kwx246/3883629/Comparison-of-Sociodemographic-and-Health-Related">which is likely in many cases</a>). The estimated <strong>observed-scale heritability</strong> is: 0.0500 (se=0.00327).</p>
<hr />
<h4 id="confounding-and-model-misspecification"><strong>Confounding and model misspecification</strong></h4>
<p>In addition to SNP heritability, LD score regression also estimates an intercept term that indexes population stratification, other counfounding, and potential misspecification in the partitioned LD score model for the distribution of genetic effects genome-wide.</p>
<ul>
<li><strong>Intercept:</strong> 1.0347 (se=0.0117, p=0.00145)</li>
<li><strong>Mean <span class="math inline">\(\chi^2\)</span>:</strong> 1.3872</li>
<li><strong>Ratio:</strong> 0.0896 (se=0.0301)</li>
<li><strong><span class="math inline">\(\lambda_{GC}\)</span>:</strong> 1.3165</li>
</ul>
<p>Intercept values near 1 indicate little or no confounding. The reported LDSR ratio compares the intercept estimate and the mean <span class="math inline">\(\chi^2\)</span> statistic to provide a rough index for how much of the polygenic signal in the GWAS may be due to confounding rather than genetic effects (assuming the LD score model is well specified). Note that the intercept, mean <span class="math inline">\(\chi^2\)</span>, and genomic control <span class="math inline">\(\lambda_{GC}\)</span> are all expected to scale with sample size, making the ratio better suited for comparisons between different GWAS.</p>
<hr />
<h4 id="correlations"><strong>Correlations</strong></h4>
<p>Genetic correlations and phenotypic correlations comparing this phenotype to other significantly heritable phenotypes from the Neale Lab UK Biobank GWAS can be found at: <a href="https://ukbb-rg.hail.is/rg_summary_2040.html"><strong>ukbb-rg.hail.is</strong></a></p>
</div>
</div>
<p><br></p>
<div class="panel panel-primary">
<div class="panel-heading">
<h3 class="panel-title">
Partitioned Heritability
</h3>
</div>
<div class="panel-body">
<h4 id="significance-of-categorical-annotations">Significance of categorical annotations</h4>
<div class="container-fluid crosstalk-bscols">
<div class="row">
<div class="col-xs-12">
<div id="htmlwidget-8b9152ca64d7ea8e752a" style="width:100%;height:400px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-8b9152ca64d7ea8e752a">{"x":{"visdat":{"1715abc69ae4":["function () ","plotlyVisDat"]},"cur_data":"1715abc69ae4","attrs":{"1715abc69ae4":{"alpha_stroke":1,"sizes":[10,100],"spans":[1,20],"x":{},"y":{},"hoverinfo":"text","text":{},"type":"bar","name":"p-value for coefficient","inherit":true},"1715abc69ae4.1":{"alpha_stroke":1,"sizes":[10,100],"spans":[1,20],"x":{},"y":[2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079],"type":"scatter","mode":"lines","hoverinfo":"text","text":"","name":"Bonferroni significance","inherit":true},"1715abc69ae4.2":{"alpha_stroke":1,"sizes":[10,100],"spans":[1,20],"x":{},"y":[1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398],"type":"scatter","mode":"lines","hoverinfo":"text","text":"","name":"Nominal significance","inherit":true}},"layout":{"margin":{"b":150,"l":100,"t":25,"r":10},"showlegend":false,"xaxis":{"domain":[0,1],"automargin":true,"title":"Annotation","tickangle":-30,"type":"category","categoryorder":"array","categoryarray":["Coding_UCSC","Conserved_LindbladToh","CTCF_Hoffman","DGF_ENCODE","DHS_peaks_Trynka","DHS_Trynka","Enhancer_Andersson","Enhancer_Hoffman","FetalDHS_Trynka","GERP.RSsup4","H3K27ac_Hnisz","H3K27ac_PGC2","H3K4me1_peaks_Trynka","H3K4me1_Trynka","H3K4me3_peaks_Trynka","H3K4me3_Trynka","H3K9ac_peaks_Trynka","H3K9ac_Trynka","Intron_UCSC","Promoter_UCSC","PromoterFlanking_Hoffman","Repressed_Hoffman","Super_Enhancer_Vahedi","SuperEnhancer_Hnisz","TFBS_ENCODE","Transcr_Hoffman","TSS_Hoffman","Typical_Enhancer_Vahedi","UTR_3_UCSC","UTR_5_UCSC","WeakEnhancer_Hoffman"]},"yaxis":{"domain":[0,1],"automargin":true,"title":"-log10(p-value)","side":"left","range":[0,3.20754648136213]},"hovermode":"closest"},"source":"A","config":{"modeBarButtonsToAdd":[],"cloud":false,"showLink":false,"displayModeBar":"hover","displaylogo":false,"sendData":false},"data":[{"x":["Coding_UCSC","Conserved_LindbladToh","CTCF_Hoffman","DGF_ENCODE","DHS_peaks_Trynka","DHS_Trynka","Enhancer_Andersson","Enhancer_Hoffman","FetalDHS_Trynka","H3K27ac_Hnisz","H3K27ac_PGC2","H3K4me1_peaks_Trynka","H3K4me1_Trynka","H3K4me3_peaks_Trynka","H3K4me3_Trynka","H3K9ac_peaks_Trynka","H3K9ac_Trynka","Intron_UCSC","PromoterFlanking_Hoffman","Promoter_UCSC","Repressed_Hoffman","SuperEnhancer_Hnisz","TFBS_ENCODE","Transcr_Hoffman","TSS_Hoffman","UTR_3_UCSC","UTR_5_UCSC","WeakEnhancer_Hoffman","Super_Enhancer_Vahedi","Typical_Enhancer_Vahedi","GERP.RSsup4"],"y":[0.552942691298678,0.840838960122168,0.539488397019624,0.256212340154997,0.323110984127356,0.0391002876234955,0.240919226857816,0.569556349770905,0.568619990331402,0.153745729925738,0.593076613497464,1.01249719143394,0.147933064470262,0.280620388934788,0.544814715598903,0.246040266167906,0.0954107772539859,0.915911152300519,0.504436304579647,0.734582716979136,0.343683660592579,0.133022114614648,0.0686201369349902,0.226738633378461,0.564240651629664,0.949351747018562,0.288932612140322,0.798825395230849,0.499713581819895,0.328186538393997,3.20754648136213],"hoverinfo":["text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text"],"text":["Annotation: Coding_UCSC<br>p-value: 0.28","Annotation: Conserved_LindbladToh<br>p-value: 0.144","Annotation: CTCF_Hoffman<br>p-value: 0.289","Annotation: DGF_ENCODE<br>p-value: 0.554","Annotation: DHS_peaks_Trynka<br>p-value: 0.475","Annotation: DHS_Trynka<br>p-value: 0.914","Annotation: Enhancer_Andersson<br>p-value: 0.574","Annotation: Enhancer_Hoffman<br>p-value: 0.269","Annotation: FetalDHS_Trynka<br>p-value: 0.27","Annotation: H3K27ac_Hnisz<br>p-value: 0.702","Annotation: H3K27ac_PGC2<br>p-value: 0.255","Annotation: H3K4me1_peaks_Trynka<br>p-value: 0.0972","Annotation: H3K4me1_Trynka<br>p-value: 0.711","Annotation: H3K4me3_peaks_Trynka<br>p-value: 0.524","Annotation: H3K4me3_Trynka<br>p-value: 0.285","Annotation: H3K9ac_peaks_Trynka<br>p-value: 0.567","Annotation: H3K9ac_Trynka<br>p-value: 0.803","Annotation: Intron_UCSC<br>p-value: 0.121","Annotation: PromoterFlanking_Hoffman<br>p-value: 0.313","Annotation: Promoter_UCSC<br>p-value: 0.184","Annotation: Repressed_Hoffman<br>p-value: 0.453","Annotation: SuperEnhancer_Hnisz<br>p-value: 0.736","Annotation: TFBS_ENCODE<br>p-value: 0.854","Annotation: Transcr_Hoffman<br>p-value: 0.593","Annotation: TSS_Hoffman<br>p-value: 0.273","Annotation: UTR_3_UCSC<br>p-value: 0.112","Annotation: UTR_5_UCSC<br>p-value: 0.514","Annotation: WeakEnhancer_Hoffman<br>p-value: 0.159","Annotation: Super_Enhancer_Vahedi<br>p-value: 0.316","Annotation: Typical_Enhancer_Vahedi<br>p-value: 0.47","Annotation: GERP.RSsup4<br>p-value: 0.00062"],"type":"bar","name":"p-value for coefficient","marker":{"color":"rgba(31,119,180,1)","line":{"color":"rgba(31,119,180,1)"}},"error_y":{"color":"rgba(31,119,180,1)"},"error_x":{"color":"rgba(31,119,180,1)"},"xaxis":"x","yaxis":"y","frame":null},{"x":["Coding_UCSC","Conserved_LindbladToh","CTCF_Hoffman","DGF_ENCODE","DHS_peaks_Trynka","DHS_Trynka","Enhancer_Andersson","Enhancer_Hoffman","FetalDHS_Trynka","H3K27ac_Hnisz","H3K27ac_PGC2","H3K4me1_peaks_Trynka","H3K4me1_Trynka","H3K4me3_peaks_Trynka","H3K4me3_Trynka","H3K9ac_peaks_Trynka","H3K9ac_Trynka","Intron_UCSC","PromoterFlanking_Hoffman","Promoter_UCSC","Repressed_Hoffman","SuperEnhancer_Hnisz","TFBS_ENCODE","Transcr_Hoffman","TSS_Hoffman","UTR_3_UCSC","UTR_5_UCSC","WeakEnhancer_Hoffman","Super_Enhancer_Vahedi","Typical_Enhancer_Vahedi","GERP.RSsup4"],"y":[2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079],"type":"scatter","mode":"lines","hoverinfo":["text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text"],"text":["","","","","","","","","","","","","","","","","","","","","","","","","","","","","","",""],"name":"Bonferroni significance","marker":{"color":"rgba(255,127,14,1)","line":{"color":"rgba(255,127,14,1)"}},"error_y":{"color":"rgba(255,127,14,1)"},"error_x":{"color":"rgba(255,127,14,1)"},"line":{"color":"rgba(255,127,14,1)"},"xaxis":"x","yaxis":"y","frame":null},{"x":["Coding_UCSC","Conserved_LindbladToh","CTCF_Hoffman","DGF_ENCODE","DHS_peaks_Trynka","DHS_Trynka","Enhancer_Andersson","Enhancer_Hoffman","FetalDHS_Trynka","H3K27ac_Hnisz","H3K27ac_PGC2","H3K4me1_peaks_Trynka","H3K4me1_Trynka","H3K4me3_peaks_Trynka","H3K4me3_Trynka","H3K9ac_peaks_Trynka","H3K9ac_Trynka","Intron_UCSC","PromoterFlanking_Hoffman","Promoter_UCSC","Repressed_Hoffman","SuperEnhancer_Hnisz","TFBS_ENCODE","Transcr_Hoffman","TSS_Hoffman","UTR_3_UCSC","UTR_5_UCSC","WeakEnhancer_Hoffman","Super_Enhancer_Vahedi","Typical_Enhancer_Vahedi","GERP.RSsup4"],"y":[1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398],"type":"scatter","mode":"lines","hoverinfo":["text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text"],"text":["","","","","","","","","","","","","","","","","","","","","","","","","","","","","","",""],"name":"Nominal significance","marker":{"color":"rgba(44,160,44,1)","line":{"color":"rgba(44,160,44,1)"}},"error_y":{"color":"rgba(44,160,44,1)"},"error_x":{"color":"rgba(44,160,44,1)"},"line":{"color":"rgba(44,160,44,1)"},"xaxis":"x","yaxis":"y","frame":null}],"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</div>
<p>Two-sided test of the tau coefficient for the named annotation conditional on the rest of the baseline-LD v1.1 model. The orange line indicates Bonferroni-corrected significance across 38 annotations of interest for this phenotype (<span class="math inline">\(p &lt;\)</span> 1.3e-3). The green line indicates nominal significance (<span class="math inline">\(p &lt;\)</span> .05). Bonferroni correction excludes the base annotation, 10 MAF bin annotations, and the 26 buffer annotations (extending 500-bp around the baseline functional annotations) since they are primarily intended as statistical controls rather than for direct inference. Note this does not account for multiple testing across phenotypes.</p>
<p><br></p>
<h4 id="significance-of-continuous-annotations">Significance of continuous annotations</h4>
<div class="container-fluid crosstalk-bscols">
<div class="row">
<div class="col-xs-12">
<div id="htmlwidget-e622aa070c7ba4097064" style="width:100%;height:400px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-e622aa070c7ba4097064">{"x":{"visdat":{"1715a2f7db8fa":["function () ","plotlyVisDat"]},"cur_data":"1715a2f7db8fa","attrs":{"1715a2f7db8fa":{"alpha_stroke":1,"sizes":[10,100],"spans":[1,20],"x":{},"y":{},"hoverinfo":"text","text":{},"type":"bar","name":"p-value for coefficient","inherit":true},"1715a2f7db8fa.1":{"alpha_stroke":1,"sizes":[10,100],"spans":[1,20],"x":{},"y":[2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079],"type":"scatter","mode":"lines","hoverinfo":"text","text":"","name":"Bonferroni significance","inherit":true},"1715a2f7db8fa.2":{"alpha_stroke":1,"sizes":[10,100],"spans":[1,20],"x":{},"y":[1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398],"type":"scatter","mode":"lines","hoverinfo":"text","text":"","name":"Nominal significance","inherit":true}},"layout":{"margin":{"b":150,"l":100,"t":25,"r":10},"showlegend":false,"xaxis":{"domain":[0,1],"automargin":true,"title":"Annotation","tickangle":-30,"type":"category","categoryorder":"array","categoryarray":["Backgrd_Selection_Stat","CpG_Content_50kb","GERP.NS","MAF_Adj_LLD_AFR","MAF_Adj_Predicted_Allele_Age","Nucleotide_Diversity_10kb","Recomb_Rate_10kb"]},"yaxis":{"domain":[0,1],"automargin":true,"title":"-log10(p-value)","side":"left","range":[0,8.81879305734063]},"hovermode":"closest"},"source":"A","config":{"modeBarButtonsToAdd":[],"cloud":false,"showLink":false,"displayModeBar":"hover","displaylogo":false,"sendData":false},"data":[{"x":["GERP.NS","MAF_Adj_Predicted_Allele_Age","MAF_Adj_LLD_AFR","Recomb_Rate_10kb","Nucleotide_Diversity_10kb","Backgrd_Selection_Stat","CpG_Content_50kb"],"y":[8.54940745158777,1.10284070523887,2.64105911859248,5.12874212763188,0.668639957910951,0.706611518458352,8.81879305734063],"hoverinfo":["text","text","text","text","text","text","text"],"text":["Annotation: GERP.NS<br>p-value: 2.82e-09","Annotation: MAF_Adj_Predicted_Allele_Age<br>p-value: 0.0789","Annotation: MAF_Adj_LLD_AFR<br>p-value: 0.00229","Annotation: Recomb_Rate_10kb<br>p-value: 7.43e-06","Annotation: Nucleotide_Diversity_10kb<br>p-value: 0.214","Annotation: Backgrd_Selection_Stat<br>p-value: 0.197","Annotation: CpG_Content_50kb<br>p-value: 1.52e-09"],"type":"bar","name":"p-value for coefficient","marker":{"color":"rgba(31,119,180,1)","line":{"color":"rgba(31,119,180,1)"}},"error_y":{"color":"rgba(31,119,180,1)"},"error_x":{"color":"rgba(31,119,180,1)"},"xaxis":"x","yaxis":"y","frame":null},{"x":["GERP.NS","MAF_Adj_Predicted_Allele_Age","MAF_Adj_LLD_AFR","Recomb_Rate_10kb","Nucleotide_Diversity_10kb","Backgrd_Selection_Stat","CpG_Content_50kb"],"y":[2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079,2.88081359228079],"type":"scatter","mode":"lines","hoverinfo":["text","text","text","text","text","text","text"],"text":["","","","","","",""],"name":"Bonferroni significance","marker":{"color":"rgba(255,127,14,1)","line":{"color":"rgba(255,127,14,1)"}},"error_y":{"color":"rgba(255,127,14,1)"},"error_x":{"color":"rgba(255,127,14,1)"},"line":{"color":"rgba(255,127,14,1)"},"xaxis":"x","yaxis":"y","frame":null},{"x":["GERP.NS","MAF_Adj_Predicted_Allele_Age","MAF_Adj_LLD_AFR","Recomb_Rate_10kb","Nucleotide_Diversity_10kb","Backgrd_Selection_Stat","CpG_Content_50kb"],"y":[1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398,1.30102999566398],"type":"scatter","mode":"lines","hoverinfo":["text","text","text","text","text","text","text"],"text":["","","","","","",""],"name":"Nominal significance","marker":{"color":"rgba(44,160,44,1)","line":{"color":"rgba(44,160,44,1)"}},"error_y":{"color":"rgba(44,160,44,1)"},"error_x":{"color":"rgba(44,160,44,1)"},"line":{"color":"rgba(44,160,44,1)"},"xaxis":"x","yaxis":"y","frame":null}],"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</div>
<p>Two-sided test of the tau coefficient for the named annotation conditional on the rest of the baseline-LD v1.1 model. The orange line indicates Bonferroni-corrected significance across 38 annotations of interest for this phenotype (<span class="math inline">\(p &lt;\)</span> 1.3e-3). The green line indicates nominal significance (<span class="math inline">\(p &lt;\)</span> .05). Bonferroni correction excludes the base annotation, 10 MAF bin annotations, and the 26 buffer annotations (extending 500-bp around the baseline functional annotations) since they are primarily intended as statistical controls rather than for direct inference. Note this does not account for multiple testing across phenotypes.</p>
</div>
</div>
<p><br></p>
<div class="well">
<h4 id="methods">Methods</h4>
<p>All results are from partitioned heritability analysis of this phenotype using LD score regression (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4495769/">Bulik-Sullivan et al. 2015</a>, <a href="https://github.com/bulik/ldsc">github repo</a>) with 75 annotations as described by <a href="http://www.nature.com/ng/journal/vaop/ncurrent/full/ng.3954.html">Gazal et al. 2017</a> (also on <a href="http://www.biorxiv.org/content/early/2017/04/03/082024">biorxiv</a>). See <a href="%60methods.html%60">Methods</a> for more information on the underlying GWAS and LDSR analysis. You can also read more about the <a href="confidence.html">confidence criteria</a> and the <a href="significance.html">significance thresholds</a>.</p>
<p>
</p>
<h4 id="downloads">Downloads</h4>
<ul>
<li><strong>Full GWAS results (large file):</strong> <a href='https://broad-ukb-sumstats-us-east-1.s3.amazonaws.com/round2/additive-tsvs/2040.gwas.imputed_v3.both_sexes.tsv.bgz'  onclick="trackOutboundLink('https://broad-ukb-sumstats-us-east-1.s3.amazonaws.com/round2/additive-tsvs/2040.gwas.imputed_v3.both_sexes.tsv.bgz'); return false;"><code>2040.gwas.imputed_v3.both_sexes.tsv.bgz</code></a></li>
<li><strong>LDSR-formatted summary statistics:</strong> <a href='https://www.dropbox.com/s/qn4wkib5f0gl94z/2040.ldsc.imputed_v3.both_sexes.tsv.bgz?dl=1'  onclick="trackOutboundLink('https://www.dropbox.com/s/qn4wkib5f0gl94z/2040.ldsc.imputed_v3.both_sexes.tsv.bgz?dl=1'); return false;"><code>2040.ldsc.imputed_v3.both_sexes.tsv.bgz</code></a></li>
<li><strong>LDSR <span class="math inline">\(h^2_g\)</span> and intercepts for all primary GWAS:</strong> <a href="https://www.dropbox.com/s/8vca84rsslgbsua/ukb31063_h2_topline.02Oct2019.tsv.gz?dl=1" onclick="trackOutboundLink('https://www.dropbox.com/s/8vca84rsslgbsua/ukb31063_h2_topline.02Oct2019.tsv.gz?dl=1'); return false;"><code>ukb31063_h2_topline.02Oct2019.tsv.gz</code></a></li>
<li><strong>LDSR partitioning for significantly heritable phenotypes:</strong> <a href="https://www.dropbox.com/s/gu9ueqdulze9c9x/ukb31063_h2_z4.02Oct2019.tsv.gz?dl=1" onclick="trackOutboundLink('https://www.dropbox.com/s/gu9ueqdulze9c9x/ukb31063_h2_z4.02Oct2019.tsv.gz?dl=1'); return false;"><code>ukb31063_h2_z4.02Oct2019.tsv.gz</code></a></li>
</ul>
<p>See the <a href="https://docs.google.com/spreadsheets/d/1EmwlCYYqkoVKqAS71nNDKoN18PyKwysYUppKTSMvKiM/edit?usp=sharing" onclick="trackOutboundLink('https://docs.google.com/spreadsheets/d/1EmwlCYYqkoVKqAS71nNDKoN18PyKwysYUppKTSMvKiM/edit?usp=sharing'); return false;">full manifest of LDSR sumstats files</a> for sumstats for other GWAS of this phenotype, where applicable.</p>
<p>See the <a href="downloads.html">Downloads page</a> for more information.</p>
<p>
</p>
<h4 id="credits">Credits</h4>
<p>See the full team behind these results <a href="credits.html">here</a>.</p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
