<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Heritability of &gt;4,000 traits &amp; disorders in UK Biobank</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<!-- setup favicons -->
<link rel="apple-touch-icon-precomposed" sizes="57x57" href="icon/apple-touch-icon-57x57.png" />
<link rel="apple-touch-icon-precomposed" sizes="114x114" href="icon/apple-touch-icon-114x114.png" />
<link rel="apple-touch-icon-precomposed" sizes="72x72" href="icon/apple-touch-icon-72x72.png" />
<link rel="apple-touch-icon-precomposed" sizes="144x144" href="icon/apple-touch-icon-144x144.png" />
<link rel="apple-touch-icon-precomposed" sizes="60x60" href="icon/apple-touch-icon-60x60.png" />
<link rel="apple-touch-icon-precomposed" sizes="120x120" href="icon/apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon-precomposed" sizes="76x76" href="icon/apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon-precomposed" sizes="152x152" href="icon/apple-touch-icon-152x152.png" />
<link rel="icon" type="image/png" href="icon/favicon-196x196.png" sizes="196x196" />
<link rel="icon" type="image/png" href="icon/favicon-96x96.png" sizes="96x96" />
<link rel="icon" type="image/png" href="icon/favicon-32x32.png" sizes="32x32" />
<link rel="icon" type="image/png" href="icon/favicon-16x16.png" sizes="16x16" />
<link rel="icon" type="image/png" href="icon/favicon-128.png" sizes="128x128" />
<meta name="application-name" content="&nbsp;"/>
<meta name="msapplication-TileColor" content="#FFFFFF" />
<meta name="msapplication-TileImage" content="icon/mstile-144x144.png" />
<meta name="msapplication-square70x70logo" content="icon/mstile-70x70.png" />
<meta name="msapplication-square150x150logo" content="icon/mstile-150x150.png" />
<meta name="msapplication-wide310x150logo" content="icon/mstile-310x150.png" />
<meta name="msapplication-square310x310logo" content="icon/mstile-310x310.png" />


<!-- Basic analytics to track views and downloads -->
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-106608402-1', 'auto');
  ga('set', 'anonymizeIp', true);
  ga('send', 'pageview');

</script>

<script>
/**
* Function that tracks a click on an outbound link in Analytics.
# modifications from: https://hacks.mozilla.org/2016/01/google-analytics-privacy-and-event-tracking/
*/
var trackOutboundLink = function(url) {
    // check if the GA object exists and that it has initialized
    if(typeof ga !== 'undefined' && ga.loaded) {
        // if yes, rely on GA to follow link
        ga('send', 'event', 'outbound', 'click', url, {
            'transport': 'beacon',
            'hitCallback': function(){document.location = url;}
        });
    } else {
        // if not, follow link ourselves
        document.location = url;
    }
}
</script>




<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="sandflat.min.css" type="text/css" />
<link rel="stylesheet" href="dt_style.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">UKB Heritability</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="h2_browser.html">Browser</a>
</li>
<li>
  <a href="downloads.html">Downloads</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Methods
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="details.html">Overview</a>
    </li>
    <li>
      <a href="select_topline.html">Selecting Primary GWAS</a>
    </li>
    <li>
      <a href="confidence.html">Confidence Criteria</a>
    </li>
    <li>
      <a href="significance.html">Significance Thresholds</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Plots
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="viz_h2.html">Heritability and Intercept</a>
    </li>
    <li>
      <a href="viz_annot.html">Partitioned Annotations</a>
    </li>
    <li>
      <a href="viz_sampsize.html">Sample Size Scaling</a>
    </li>
    <li>
      <a href="viz_R1vR2.html">Round 1 vs. Round 2</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Archive
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="round1_details.html">Round 1 Methods</a>
    </li>
    <li>
      <a href="round1_h2_browser.html">Round 1 Browser</a>
    </li>
    <li>
      <a href="round1_h2_univar_browser.html">Round 1 Univariate Browser</a>
    </li>
    <li>
      <a href="round1_plots_home.html">Round 1 Plots</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    About
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="credits.html">Credits</a>
    </li>
    <li>
      <a href="https://github.com/Nealelab/UKBB_ldsc">Site Github</a>
    </li>
    <li>
      <a href="https://github.com/Nealelab/UKBB_ldsc_scripts">Analysis Github</a>
    </li>
    <li>
      <a href="https://www.nealelab.is/uk-biobank">Neale Lab UKB</a>
    </li>
  </ul>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Heritability of &gt;4,000 traits &amp; disorders in UK Biobank</h1>

</div>


<p><br></p>
<div class="well">
<p>As you may know, the UK Biobank released data on ~500,000 individuals to the scientific community to advance research on genetics. The <a href="http://nealelab.is">Neale Lab</a> has been working to make some basic analyses of that data freely available; you can read more about the overall effort <a href="http://www.nealelab.is/uk-biobank">here</a>.</p>
<p>
</p>
<p>Building on the initial genome-wide association analyses, we’ve now estimated the heritability of each of those traits and disorders, over 4,000 as of this writing. As with the primary GWAS analyses, we’re making the LD Score Regression SNP-heritability results available for <a href="h2_browser.html">browsing</a> and bulk <a href="downloads.html">download</a>.</p>
<p>
</p>
<p>If you’re interested in additional traits, additional ancestry groups, and/or other heritability estimation methods, SNP hertiability estimates are also now available from the <a href="https://pan.ukbb.broadinstitute.org/docs/heritability">Pan-UK Biobank analysis</a>.</p>
<p><br></p>
<p><a href="h2_browser.html" class="btn btn-primary btn-lg">Browse <span class="math inline">\(h^2\)</span></a> <a href="viz_h2.html" class="btn btn-primary btn-lg">Explore Plots</a> <a href="downloads.html"  class="btn btn-primary btn-lg">Download Results</a></p>
<p><br></p>
<p>You can learn more about the Round 2 Neale Lab GWAS heritability analysis in this series of blog posts:</p>
<ul>
<li><a href="http://www.nealelab.is/blog/2017/9/13/heritability-101-what-is-heritability">Heritability 101: What is “heritability”?</a></li>
<li><a href="http://www.nealelab.is/blog/2017/9/13/heritability-201-types-of-heritability-and-how-we-estimate-it">Heritability 201: Types of heritability and how we estimate it</a></li>
<li><a href="http://www.nealelab.is/blog/2017/9/20/insights-from-estimates-of-snp-heritability-for-2000-traits-and-disorders-in-uk-biobank">Insights from estimates of SNP-heritability for &gt;2,000 traits and disorders in UK Biobank</a></li>
</ul>
<p>
</p>
<p>More technical descriptions are also available:</p>
<ul>
<li><a href="http://www.nealelab.is/blog/2017/9/14/heritability-501-ldsr-based-h2-in-ukbb-for-the-technically-minded">Heritability 501: LDSR-based <span class="math inline">\(h^2\)</span> in UKBB for the technically-minded</a></li>
<li><a href="https://github.com/Nealelab/UKBB_ldsc_scripts">UKBB_ldsc_scripts github repository</a></li>
</ul>
<p>
</p>
<p>We view these results as preliminary and subject to change. They rely on uniform automated GWAS of thousands of variables, each of which can potentially be refined by subject matter experts. You can <a href="viz_R1vR2.html">view the change between our Round 1 and Round 2 results here</a>. But we hope you’ll agree that the current results still provide an interesting and useful look at the genetics of the many diverse traits and disorders studied in the UK Biobank, and are potentially informative for highlighting phenotypes meriting further follow-up.</p>
<p>
</p>
<p>If you want more information, please <strong>contact us at: <a href="mailto:nealelab.ukb@gmail.com">nealelab.ukb@gmail.com</a>.</strong></p>
</div>
<p><br></p>
<div class="well">
<h3 id="faq">FAQ</h3>
<h4 id="where-can-i-find-the-heritability-results"><strong>Where can I find the heritability results?</strong></h4>
<p>We’ve created a browser to explore the results <a href="h2_browser.html">here</a>. You can also download the full results file <a href="https://www.dropbox.com/s/ipeqyhrpdqav5uh/ukb31063_h2_all.02Oct2019.tsv.gz?dl=1" onclick="trackOutboundLink('https://www.dropbox.com/s/ipeqyhrpdqav5uh/ukb31063_h2_all.02Oct2019.tsv.gz?dl=1'); return false;">here</a>.</p>
<p>
</p>
<h4 id="what-method-have-you-used-to-estimated-heritability"><strong>What method have you used to estimated heritability?</strong></h4>
<p>To enable this quick initial analysis on such large-scale data we’re relying on LD score regression (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4495769/">article</a>, <a href="https://github.com/bulik/ldsc">github repo</a>). Full code and technical details are available in the <a href="https://github.com/Nealelab/UKBB_ldsc_scripts">UKBB_ldsc_scripts github repository</a> and described in <a href="http://www.nealelab.is/blog/2017/9/14/heritability-501-ldsr-based-h2-in-ukbb-for-the-technically-minded">Heritability 501</a>.</p>
<p>
</p>
<h4 id="are-the-primary-gwas-results-used-for-this-analysis-available"><strong>Are the primary GWAS results used for this analysis available? </strong></h4>
<p>Yes! The full GWAS results are <a href="https://docs.google.com/spreadsheets/d/1kvPoupSzsSFBNSztMzl04xMoSC3Kcx3CrjVf4yBmESU/edit#gid=227859291">available for download</a>. See <a href="http://www.nealelab.is/uk-biobank">the Neale Lab blog</a> for much more information.</p>
<p>
</p>
<h4 id="can-i-download-just-the-ldsc-formatted-sumstats-files-instead-of-the-full-gwas-results"><strong>Can I download just the LDSC-formatted sumstats files instead of the full GWAS results?</strong></h4>
<p>Yes! Direct download links for the LDSC sumstats files for all of the primary GWAS are available in a searchable table on the <a href="downloads.html#ldsc_sumstat_files">Downloads page</a>. You can also view and download the <a href="https://www.dropbox.com/s/4fantvoxkzdhcou/ukb31063_ldsc_sumstat_manifest_aws_sep2022.tsv.gz?dl=0" onclick="trackOutboundLink('https://www.dropbox.com/s/4fantvoxkzdhcou/ukb31063_ldsc_sumstat_manifest_aws_sep2022.tsv.gz?dl=0'); return false;">full manifest of ldsc sumstats files</a> for easier programmatic access and links to sumstats files for all of the Neale UKB GWAS.</p>
<p>
</p>
<h4 id="are-there-heritability-results-available-for-non-european-ancestries"><strong>Are there heritability results available for non-European ancestries?</strong></h4>
<p>Yes, the <a href="https://pan.ukbb.broadinstitute.org/">Pan-UK Biobank</a> analysis includes multiple forms of heritability analysis in 6 ancestry groups in UK Biobank. Details on the analysis can be found <a href="https://pan.ukbb.broadinstitute.org/docs/heritability">here</a>, and results are avalable <a href="https://pan.ukbb.broadinstitute.org/downloads">for download</a> or can be searched in a <a href="https://docs.google.com/spreadsheets/d/1AeeADtT0U1AukliiNyiVzVRdLYPkTbruQSk38DeutU8/edit#gid=1797288938">google sheets manifest</a>.</p>
<p>
</p>
<h4 id="how-should-i-interpret-these-heritability-results"><strong>How should I interpret these heritability results?</strong></h4>
<p><em>Very carefully.</em> For starters, we’ve written a couple of primers that cover the concept of heritability and common misconceptions (<a href="http://www.nealelab.is/blog/2017/9/13/heritability-101-what-is-heritability">Heritability 101</a>) and the specific type of heritability that’s being estimated in this analysis (<a href="http://www.nealelab.is/blog/2017/9/13/heritability-201-types-of-heritability-and-how-we-estimate-it">Heritability 201</a>). A longer discussion of the many limitations of this analysis is included in our more technical description of the project (<a href="http://www.nealelab.is/blog/2017/9/14/heritability-501-ldsr-based-h2-in-ukbb-for-the-technically-minded">Heritability 501</a>).</p>
<p>This is a very rough initial analysis and should be treated accordingly. All estimates are for (a) the heritability from common genetic variants (b) of automatically cleaned and transformed phenotypes (c) in a non-random population sample of the UK (d) estimated with a fairly simple statistical model. We expect the results will mostly be useful for generating hypotheses for follow-up (e.g. identifying trends in the estimates for categories of phenotypes, choosing phenotypes with robust statistical evidence for heritability), rather than being a definitive statement about the “true” heritability of a individual phenotype.</p>
<p>
</p>
<h4 id="are-genetic-correlation-results-available-for-these-phenotypes"><strong>Are genetic correlation results available for these phenotypes?</strong></h4>
<p>Yes! You can browse genetic correlations between the significantly heritable phenotypes at <a href="https://ukbb-rg.hail.is/" class="uri">https://ukbb-rg.hail.is/</a>. You can also <a href="downloads.html#ldsc_sumstat_files">download the LDSC-formatted sumstats files</a> for your own genetic correlation analyses.</p>
</div>
<p><br></p>
<div class="well">
<h3 id="credits">Credits</h3>
<ul>
<li><strong>Lead:</strong> Raymond Walters</li>
<li><strong>LDSR analysis support:</strong> Nikolas Baya, Katherine Tashman, Danfeng Chen, Liam Abbott</li>
<li><strong>Phenotype analysis support</strong>: Caitlin Carey, Duncan Palmer</li>
<li><strong>Supervision</strong>: Benjamin Neale</li>
<li><strong>UK Biobank GWAS Core Team:</strong> Liam Abbott, Sam Bryant, Claire Churchhouse, Andrea Ganna, Daniel Howrigan, Duncan Palmer, Ben Neale, Raymond Walters, Caitlin Carey, The Hail team, Benjamin Neale</li>
</ul>
<p><strong>Round 2 release:</strong> <a href="https://zenodo.org/badge/latestdoi/99861592"><img src="https://zenodo.org/badge/99861592.svg" alt="DOI"></a></p>
<p>See <a href="credits.html">Credits</a> for complete acknowledgements.</p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
