<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Results from the Neale Lab" />


<title>Details on LD Score Regression in UK Biobank</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<!-- setup favicons -->
<link rel="apple-touch-icon-precomposed" sizes="57x57" href="icon/apple-touch-icon-57x57.png" />
<link rel="apple-touch-icon-precomposed" sizes="114x114" href="icon/apple-touch-icon-114x114.png" />
<link rel="apple-touch-icon-precomposed" sizes="72x72" href="icon/apple-touch-icon-72x72.png" />
<link rel="apple-touch-icon-precomposed" sizes="144x144" href="icon/apple-touch-icon-144x144.png" />
<link rel="apple-touch-icon-precomposed" sizes="60x60" href="icon/apple-touch-icon-60x60.png" />
<link rel="apple-touch-icon-precomposed" sizes="120x120" href="icon/apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon-precomposed" sizes="76x76" href="icon/apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon-precomposed" sizes="152x152" href="icon/apple-touch-icon-152x152.png" />
<link rel="icon" type="image/png" href="icon/favicon-196x196.png" sizes="196x196" />
<link rel="icon" type="image/png" href="icon/favicon-96x96.png" sizes="96x96" />
<link rel="icon" type="image/png" href="icon/favicon-32x32.png" sizes="32x32" />
<link rel="icon" type="image/png" href="icon/favicon-16x16.png" sizes="16x16" />
<link rel="icon" type="image/png" href="icon/favicon-128.png" sizes="128x128" />
<meta name="application-name" content="&nbsp;"/>
<meta name="msapplication-TileColor" content="#FFFFFF" />
<meta name="msapplication-TileImage" content="icon/mstile-144x144.png" />
<meta name="msapplication-square70x70logo" content="icon/mstile-70x70.png" />
<meta name="msapplication-square150x150logo" content="icon/mstile-150x150.png" />
<meta name="msapplication-wide310x150logo" content="icon/mstile-310x150.png" />
<meta name="msapplication-square310x310logo" content="icon/mstile-310x310.png" />


<!-- Basic analytics to track views and downloads -->
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-106608402-1', 'auto');
  ga('set', 'anonymizeIp', true);
  ga('send', 'pageview');

</script>

<script>
/**
* Function that tracks a click on an outbound link in Analytics.
# modifications from: https://hacks.mozilla.org/2016/01/google-analytics-privacy-and-event-tracking/
*/
var trackOutboundLink = function(url) {
    // check if the GA object exists and that it has initialized
    if(typeof ga !== 'undefined' && ga.loaded) {
        // if yes, rely on GA to follow link
        ga('send', 'event', 'outbound', 'click', url, {
            'transport': 'beacon',
            'hitCallback': function(){document.location = url;}
        });
    } else {
        // if not, follow link ourselves
        document.location = url;
    }
}
</script>




<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="sandflat.min.css" type="text/css" />
<link rel="stylesheet" href="dt_style.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">UKB Heritability</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="h2_browser.html">Browser</a>
</li>
<li>
  <a href="downloads.html">Downloads</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Methods
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="details.html">Overview</a>
    </li>
    <li>
      <a href="select_topline.html">Selecting Primary GWAS</a>
    </li>
    <li>
      <a href="confidence.html">Confidence Criteria</a>
    </li>
    <li>
      <a href="significance.html">Significance Thresholds</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Plots
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="viz_h2.html">Heritability and Intercept</a>
    </li>
    <li>
      <a href="viz_annot.html">Partitioned Annotations</a>
    </li>
    <li>
      <a href="viz_sampsize.html">Sample Size Scaling</a>
    </li>
    <li>
      <a href="viz_R1vR2.html">Round 1 vs. Round 2</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Archive
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="round1_details.html">Round 1 Methods</a>
    </li>
    <li>
      <a href="round1_h2_browser.html">Round 1 Browser</a>
    </li>
    <li>
      <a href="round1_h2_univar_browser.html">Round 1 Univariate Browser</a>
    </li>
    <li>
      <a href="round1_plots_home.html">Round 1 Plots</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    About
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="credits.html">Credits</a>
    </li>
    <li>
      <a href="https://github.com/Nealelab/UKBB_ldsc">Site Github</a>
    </li>
    <li>
      <a href="https://github.com/Nealelab/UKBB_ldsc_scripts">Analysis Github</a>
    </li>
    <li>
      <a href="https://www.nealelab.is/uk-biobank">Neale Lab UKB</a>
    </li>
  </ul>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Details on LD Score Regression in UK Biobank</h1>
<h4 class="author"><em>Results from the <a href="credits.html">Neale Lab</a></em></h4>
<h4 class="date"><em>Last updated 2019-10-19</em></h4>

</div>


<!-- overwrite tocify scrollto to allow for header -->
<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 100
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>
<!-- CSS for code snippets, since Rmd seems to override the main CSS -->
<style>
code{
    color: #c7254e;
    background-color: #f9f2f4;
}
</style>
<p><br></p>
<div class="well">
<h3 id="methods-outline">Methods Outline</h3>
<ul>
<li><p>Genotyping and phenotyping of ~500,000 individuals performed by the <a href="https://www.ukbiobank.ac.uk/">UK Biobank</a>. Accessed under application 31063.</p></li>
<li><p>Phenotypes processed for GWAS using <a href="https://github.com/astheeggeggs/PHESANT">PHESANT</a> and by extracting ICD10 codes (letter + digits) and recreating <a href="https://www.finngen.fi/en">FinnGen</a> endpoints.</p></li>
<li><p>GWAS performed by the <a href="https://www.nealelab.is/uk-biobank">Neale Lab</a> using <a href="https://hail.is">Hail</a> on the <a href="https://cloud.google.com/">Google Cloud Platform</a>. <a href="http://www.nealelab.is/blog/2017/9/14/heritability-501-ldsr-based-h2-in-ukbb-for-the-technically-minded">A full technical discussion</a> on the limitations of this analysis and recommendations for intepreting the results based on the Round 1 results, as well as more general details about <a href="http://www.nealelab.is/blog/2017/9/11/details-and-considerations-of-the-uk-biobank-gwas">our analysis of UK Biobank</a>, are available elsewhere. Code is also available <a href="https://github.com/Nealelab/UK_Biobank_GWAS">on Github</a>.</p></li>
<li><p>GWAS summary statistics for HapMap3 sites passing QC (MAF &gt; 0.01, INFO &gt; 0.9, excluding multiallelics and the MHC region) in UK Biobank extracted in the proper format for <a href="https://github.com/bulik/ldsc">ldsc</a> using <a href="https://hail.is">Hail</a>.</p></li>
<li><p>SNP Heritability estimated using partitioned LD Score regression (LDSR; <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4495769/">Bulik-Sullivan et al. 2015</a>, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4626285/">Finucane et al. 2015</a>) run using the <a href="https://github.com/omeed-maghzian/mtag">MTAG</a> implementation of <a href="https://github.com/bulik/ldsc">ldsc</a> with the Baseline-LD v1.1 model (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6133304/">Gazal et al. 2017</a>) on the <a href="https://cloud.google.com/">Google Cloud Platform</a>.</p></li>
<li><p>Restricted primary SNP heritability analysis to <a href="select_topline.html">one GWAS per phenotype</a>. This includes identifying sex-specific phenotypes, choice of normalization for continuous phenotypes, evaluation of the dilution fraction for biomarker phenotypes, and pruning of <a href="https://www.finngen.fi/en">FinnGen</a> endpoints that are identical within current UKB data as detailed <a href="select_topline.html">here</a>. LDSR results for all primary GWAS are available from the <a href="h2_browser.html">SNP heritability browser</a>.</p></li>
<li><p>Results for each primary GWAS evaulated for <a href="confidence.html">confidence</a> based on power at potential biases as a function of sample size, the expected magnitude of the standard error on the SNP heritability estimate, and potential limitations to interpretation based on sex differences or ordinal encodings in phenotyping.</p></li>
<li><p>Among results with acceptable <a href="confidence.html">confidence</a>, tested <a href="significance.html">statistical significance</a> of the LDSR SNP heritability estimate accounting for the effective multiple testing burden. More detail on the evaluation of potential thresholds for statistical significance in this analysis <a href="significance.html">here</a>.</p></li>
<li><p>Partitioning results from LDSR reported for phenotypes with strong SNP heritability results (<span class="math inline">\(z &gt; 7\)</span>). See phenome-wide <a href="viz_annot.html">plots</a> and access per-phenotype results from the <a href="h2_browser.html">SNP heritability browser</a>.</p></li>
<li><p>All hertiability results are reported on the liability scale, assuming that the population prevalence matches the prevalence in the UK Biobank analysis set. The significance of the estimate is not affected by this conversion, and the point estimate is only affected for binary phenotypes. The liablity scale value may be unreliable if the outcome was limited to a subset of individuals, or if the UK Biobank study population is not representative for this phenotype (<a href=\"https://academic.oup.com/aje/article/doi/10.1093/aje/kwx246/3883629/Comparison-of-Sociodemographic-and-Health-Related\">which is likely in many cases</a>). Observed scale <span class="math inline">\(h^2_g\)</span> estimates are available on the per-phenotype results pages accessible from the <a href="h2_browser.html">SNP heritability browser</a> and are included in the <a href="downloads.html">downloadable results</a>.</p></li>
<li><p>Full code for the LDSR analysis is available <a href="https://github.com/Nealelab/UKBB_ldsc_scripts">on Github</a></p></li>
</ul>
</div>
<p>
</p>
<div class="well">
<h3 id="additional-ldsr-details">Additional LDSR Details</h3>
<ul>
<li><p>Sumstats files were extracted directly from the UKB results files rather than running <code>munge_sumstats.py</code> from <a href="https://github.com/bulik/ldsc">ldsc</a>. This includes directly extracting HapMap3 sites rather than using the precomputed <code>w_hm3.snplist</code> reference file.</p></li>
<li><p>LDSR was run using the implementation in <a href="https://github.com/omeed-maghzian/mtag">MTAG</a> because it provides a convenient interface to <a href="https://github.com/bulik/ldsc">ldsc</a> from within python rather than via the command line.</p></li>
<li><p>The partitioned LDSR analysis was run using the v1.1 of the Baseline-LD annotations described by <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6133304/">Gazal et al. 2017</a>. The Baseline-LD v1.1 model contains 75 annotations including functional regions, histone marks, GERP scores, LD-related annotations of population genetics forces, and MAF bins. LD scores were computed from 1000 Genomes Phase 3 data from European ancestry populations, along with weights and allele frequencies. The LD scores and corresponding reference files are available from <a href="https://data.broadinstitute.org/alkesgroup/LDSCORE/">the ldsc reference downloads page</a>).</p></li>
<li><p>Default settings were used with the exception of removing the maximum <span class="math inline">\(\chi^2\)</span> filter in order to avoid bias in phenotypes that have extremely large <span class="math inline">\(\chi^2\)</span> for loci with strong effect sizes in the large sample size of UKB.</p></li>
<li><p>We previously ran both univariate and partitioned heritability analyses. For the reasons written about <a href="http://www.nealelab.is/blog/2017/9/20/insights-from-estimates-of-snp-heritability-for-2000-traits-and-disorders-in-uk-biobank">here</a>, we’re currently presenting the partitioned heritability LDSR results as the primary analysis in the <a href="h2_browser.html">summary browser</a>. You can see how the current results compare to the Round 1 version <a href="viz_R1vR2.html">here</a>.</p></li>
</ul>
</div>
<p>
</p>
<div class="well">
<h3 id="site-design">Site Design</h3>
<ul>
<li><p>This site has been generated using <a href="http://rmarkdown.rstudio.com/">R Markdown</a>.</p></li>
<li><p>Interactive plots generated with with <a href="https://plot.ly/r/">plotly</a> and <a href="https://rstudio.github.io/crosstalk/">crosstalk</a>.</p></li>
<li><p>Searchable tables generated using <a href="https://rstudio.github.io/DT/">DT</a>.</p></li>
<li><p>CSS theme based on <a href="https://bootswatch.com/3/">Bootswatch</a> themes Flatly and Sandstone, modified with the assistance of <a href="https://www.bootstrap-live-customizer.com/">Bootstrap Live Customizer</a>.</p></li>
<li><p>Favicon generated with the assistance of <a href="http://www.favicomatic.com/">Favic-o-matic</a>.</p></li>
<li><p>Site hosted via <a href="https://pages.github.com/">Github Pages</a>.</p></li>
<li><p>All code for generating the site available <a href="https://github.com/nealelab/ukbb_ldsc">on Github</a>.</p></li>
</ul>
</div>
<p>
</p>
<div class="well">
<h3 id="contact">Contact</h3>
<p>If you want more information, please <strong>contact us at: <a href="mailto:nealelab.ukb@gmail.com">nealelab.ukb@gmail.com</a>.</strong></p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
