<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Results from the Neale Lab" />


<title>Details on LD Score Regression in UK Biobank</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<!-- setup favicons -->
<link rel="apple-touch-icon-precomposed" sizes="57x57" href="icon/apple-touch-icon-57x57.png" />
<link rel="apple-touch-icon-precomposed" sizes="114x114" href="icon/apple-touch-icon-114x114.png" />
<link rel="apple-touch-icon-precomposed" sizes="72x72" href="icon/apple-touch-icon-72x72.png" />
<link rel="apple-touch-icon-precomposed" sizes="144x144" href="icon/apple-touch-icon-144x144.png" />
<link rel="apple-touch-icon-precomposed" sizes="60x60" href="icon/apple-touch-icon-60x60.png" />
<link rel="apple-touch-icon-precomposed" sizes="120x120" href="icon/apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon-precomposed" sizes="76x76" href="icon/apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon-precomposed" sizes="152x152" href="icon/apple-touch-icon-152x152.png" />
<link rel="icon" type="image/png" href="icon/favicon-196x196.png" sizes="196x196" />
<link rel="icon" type="image/png" href="icon/favicon-96x96.png" sizes="96x96" />
<link rel="icon" type="image/png" href="icon/favicon-32x32.png" sizes="32x32" />
<link rel="icon" type="image/png" href="icon/favicon-16x16.png" sizes="16x16" />
<link rel="icon" type="image/png" href="icon/favicon-128.png" sizes="128x128" />
<meta name="application-name" content="&nbsp;"/>
<meta name="msapplication-TileColor" content="#FFFFFF" />
<meta name="msapplication-TileImage" content="icon/mstile-144x144.png" />
<meta name="msapplication-square70x70logo" content="icon/mstile-70x70.png" />
<meta name="msapplication-square150x150logo" content="icon/mstile-150x150.png" />
<meta name="msapplication-wide310x150logo" content="icon/mstile-310x150.png" />
<meta name="msapplication-square310x310logo" content="icon/mstile-310x310.png" />


<!-- Basic analytics to track views and downloads -->
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-106608402-1', 'auto');
  ga('set', 'anonymizeIp', true);
  ga('send', 'pageview');

</script>

<script>
/**
* Function that tracks a click on an outbound link in Analytics.
# modifications from: https://hacks.mozilla.org/2016/01/google-analytics-privacy-and-event-tracking/
*/
var trackOutboundLink = function(url) {
    // check if the GA object exists and that it has initialized
    if(typeof ga !== 'undefined' && ga.loaded) {
        // if yes, rely on GA to follow link
        ga('send', 'event', 'outbound', 'click', url, {
            'transport': 'beacon',
            'hitCallback': function(){document.location = url;}
        });
    } else {
        // if not, follow link ourselves
        document.location = url;
    }
}
</script>




<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="sandflat.min.css" type="text/css" />
<link rel="stylesheet" href="dt_style.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">UKB Heritability</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="h2_browser.html">Browser</a>
</li>
<li>
  <a href="downloads.html">Downloads</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Methods
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="details.html">Overview</a>
    </li>
    <li>
      <a href="select_topline.html">Selecting Primary GWAS</a>
    </li>
    <li>
      <a href="confidence.html">Confidence Criteria</a>
    </li>
    <li>
      <a href="significance.html">Significance Thresholds</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Plots
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="viz_h2.html">Heritability and Intercept</a>
    </li>
    <li>
      <a href="viz_annot.html">Partitioned Annotations</a>
    </li>
    <li>
      <a href="viz_sampsize.html">Sample Size Scaling</a>
    </li>
    <li>
      <a href="viz_R1vR2.html">Round 1 vs. Round 2</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Archive
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="round1_details.html">Round 1 Methods</a>
    </li>
    <li>
      <a href="round1_h2_browser.html">Round 1 Browser</a>
    </li>
    <li>
      <a href="round1_h2_univar_browser.html">Round 1 Univariate Browser</a>
    </li>
    <li>
      <a href="round1_plots_home.html">Round 1 Plots</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    About
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="credits.html">Credits</a>
    </li>
    <li>
      <a href="https://github.com/Nealelab/UKBB_ldsc">Site Github</a>
    </li>
    <li>
      <a href="https://github.com/Nealelab/UKBB_ldsc_scripts">Analysis Github</a>
    </li>
    <li>
      <a href="https://www.nealelab.is/uk-biobank">Neale Lab UKB</a>
    </li>
  </ul>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Details on LD Score Regression in UK Biobank</h1>
<h4 class="author"><em>Results from the <a href="credits.html">Neale Lab</a></em></h4>
<h4 class="date"><em>Last updated 2019-10-19</em></h4>

</div>


<!-- CSS for code snippets, since Rmd seems to override the main CSS -->
<style>
code{
    color: #c7254e;
    background-color: #f9f2f4;
}
</style>
<p><br></p>
<div class="well">
<h4 id="summary">Summary</h4>
<ul>
<li><p>SNP Heritability is estimated are from LD Score Regression (LDSR; <a href="http://www.nature.com/ng/journal/v47/n3/full/ng.3211.html">Bulik-Sullivan et al. 2015</a> or <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4495769/">on pubmed</a>) using <a href="https://github.com/bulik/ldsc">ldsc</a> and <a href="https://github.com/omeed-maghzian/mtag">MTAG</a></p></li>
<li><p>Full code is available on the <a href="https://github.com/Nealelab/UKBB_ldsc">Github repo</a></p></li>
<li><p>The <a href="http://www.nealelab.is/blog">Neale Lab blog</a> has <a href="http://www.nealelab.is/blog/2017/9/14/heritability-501-ldsr-based-h2-in-ukbb-for-the-technically-minded">a full technical discussion</a> on the limitations of this analysis and recommendations for intepreting the results, as well as more general details about <a href="http://www.nealelab.is/blog/2017/9/11/details-and-considerations-of-the-uk-biobank-gwas">our analysis of UK Biobank</a>.</p></li>
<li><p>Results on this site are very loosely filtered to report only phenotypes with effective N &gt; 200. Where appropriate, stricter filters are applied as indicated. Results for phenotypes with effective N &lt; 200 can still be found in the <a href="https://www.dropbox.com/sh/g28ee03vy4hvqw5/AADAkDbSFHsYE8bme1Jjeekca/ldsc_results?dl=0" onclick="trackOutboundLink('https://www.dropbox.com/sh/g28ee03vy4hvqw5/AADAkDbSFHsYE8bme1Jjeekca/ldsc_results?dl=0'); return false;">downloads</a>.</p></li>
</ul>
<p><br> <br></p>
<h4 id="ldsr-details">LDSR Details</h4>
<ul>
<li><p>We follow the conventional approach of restricting to GWAS results from HapMap3 sites passing MAF &gt; 0.01 and INFO &gt; 0.9 as input, but we’ve reimplemented this to define a single passing list in UKB rather than running <code>munge_sumstats.py</code> from <a href="https://github.com/bulik/ldsc">ldsc</a> with the <code>w_hm3.snplist</code> reference file.</p></li>
<li><p>We’ve run both univariate and partitioned heritability analyses. For reasons that will be discussed in a later post we’re currently presenting the partitioned heritability LDSR results as the primary analysis in the <a href="h2_browser.html">summary browser</a>, but the univariate results are also available <a href="h2_univar_browser.html">here</a>.</p></li>
<li><p>The univariate LDSR analysis was run with default settings using <a href="https://data.broadinstitute.org/alkesgroup/LDSCORE/">precomputed LD scores from 1000 Genomes European ancestry samples</a> (i.e. <code>./eur_w_ld</code>)</p></li>
<li><p>The partitioned LDSR analysis was run using the v1.1 of the Baseline-LD annotations described by <a href="http://www.nature.com/ng/journal/vaop/ncurrent/full/ng.3954.html">Gazal et al. 2017</a> (also on <a href="http://www.biorxiv.org/content/early/2017/04/03/082024">biorxiv</a>) computed from 1000 Genomes Phase 3 data from European ancestry populations and corresponding allele frequencies (available from <a href="https://data.broadinstitute.org/alkesgroup/LDSCORE/">the ldsc reference downloads page</a>). Default settings were used, with the exception of removing the maximum <span class="math inline">\(\chi^2\)</span> filter due to the extreme sample size of UK Biobank.</p></li>
</ul>
<p><br> <br></p>
<h4 id="tools">Tools</h4>
<ul>
<li><p>LDSR was run using the implementation in <a href="https://github.com/omeed-maghzian/mtag">MTAG</a>, which provides a convenient interface to <a href="https://github.com/bulik/ldsc">ldsc</a> from within python rather than via the command line.</p></li>
<li><p>All analyses were run on the <a href="https://cloud.google.com/">Google Cloud Platform</a>, with the assistance of <a href="https://hail.is">Hail</a>, <a href="https://github.com/Nealelab/cloudtools">cloudtools</a>, and many other tools.</p></li>
<li><p>This site has been generated with the assistance of <a href="http://rmarkdown.rstudio.com/">R Markdown</a> with <a href="https://plot.ly/r/">plotly</a>, <a href="https://rstudio.github.io/DT/">DT</a>, and <a href="https://rstudio.github.io/crosstalk/">crosstalk</a>, and is hosted via <a href="https://pages.github.com/">Github Pages</a>.</p></li>
</ul>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
