---
title: "Heritability of >4,000 traits & disorders in UK Biobank"
output: html_document
---

<br>

<div class="well">

As you may know, the UK Biobank released data on ~500,000 individuals to the scientific community to advance research on genetics. The [Neale Lab](http://nealelab.is) at Massachusetts General Hospital and the Broad Institute of MIT and Harvard has been working to make some basic analyses of that data freely available; you can read more about the overall effort [here](http://www.nealelab.is/uk-biobank).

<p></p>

Building on the initial genome-wide association analyses, we’ve now estimated the heritability of each of those traits and disorders, over 4,000 as of this writing. As with the primary GWAS analyses, we’re making the heritability results available for [browsing](h2_browser.html) and bulk [download](downloads.html). 

<br>

<a href="h2_browser.html" class="btn btn-primary btn-lg">Browse $h^2$</a>
<a href="plots_home.html" class="btn btn-primary btn-lg">Explore Plots</a>
<a href="downloads.html"  class="btn btn-primary btn-lg">Download Results</a>

<br>

You can learn more about the heritability analysis in this series of blog posts:

* [Heritability 101: What is “heritability”?](http://www.nealelab.is/blog/2017/9/13/heritability-101-what-is-heritability)
* [Heritability 201: Types of heritability and how we estimate it](http://www.nealelab.is/blog/2017/9/13/heritability-201-types-of-heritability-and-how-we-estimate-it)
* [Insights from estimates of SNP-heritability for >2,000 traits and disorders in UK Biobank](http://www.nealelab.is/blog/2017/9/20/insights-from-estimates-of-snp-heritability-for-2000-traits-and-disorders-in-uk-biobank)

<p></p>

More technical descriptions are also available:

* [Heritability 501: LDSR-based $h^2$ in UKBB for the technically-minded](http://www.nealelab.is/blog/2017/9/14/heritability-501-ldsr-based-h2-in-ukbb-for-the-technically-minded)
* [UKBB_ldsc_scripts github repository](https://github.com/Nealelab/UKBB_ldsc_scripts)

<p></p>

We view these results as preliminary and subject to change. They rely on uniform automated GWAS of thousands of variables, each of which can potentially be refined by subject matter experts. You can [browse the change between our Round 1 and Round 2 results here](). But we hope you’ll agree that the current results still provide an interesting and useful look at the genetics of the many diverse traits and disorders studied in the UK Biobank, and are potentially informative for highlighting phenotypes meriting further follow-up.

<p></p>

If you want more information, please **contact us at: <nealelab.ukb@gmail.com>.**

</div>

<br>

<div class="well">

### FAQ

#### **Where can I find the heritability results?**

We’ve created a browser to explore the results [here](h2_browser.html). You can also download the full results file <a href="https://www.dropbox.com/sh/g28ee03vy4hvqw5/AADAkDbSFHsYE8bme1Jjeekca/ldsc_results?dl=0" onclick="trackOutboundLink('https://www.dropbox.com/sh/g28ee03vy4hvqw5/AADAkDbSFHsYE8bme1Jjeekca/ldsc_results?dl=0'); return false;">here</a>.

<p></p>

#### **What method have you used to estimated heritability?**

To enable this quick initial analysis on such large-scale data we’re relying on LD score regression ([article](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4495769/), [github repo](https://github.com/bulik/ldsc)). Full code and technical details are available in the [UKBB_ldsc_scripts github repository](https://github.com/Nealelab/UKBB_ldsc_scripts) and described in [Heritability 501](http://www.nealelab.is/blog/2017/9/14/heritability-501-ldsr-based-h2-in-ukbb-for-the-technically-minded).

<p></p>

#### **Are the primary GWAS results used for this analysis available? **

Yes! The full GWAS results are [available for download](https://docs.google.com/spreadsheets/d/1kvPoupSzsSFBNSztMzl04xMoSC3Kcx3CrjVf4yBmESU/edit#gid=227859291). See [the Neale Lab blog](http://www.nealelab.is/uk-biobank) for much more information. 

<p></p>

#### **Can I download just the LDSC-formatted sumstats files without dealing with the full GWAS results?**

Yes! Sumstats files are available [on Dropbox](). See [this manifest]() for easy searching and programmatic access to the file downloads. 

<p></p>

#### **How should I interpret these heritability results?**

*Very carefully.* For starters, we’ve written a couple of primers that cover the concept of heritability and common misconceptions ([Heritability 101](http://www.nealelab.is/blog/2017/9/13/heritability-101-what-is-heritability)) and the specific type of heritability that’s being estimated in this analysis ([Heritability 201](http://www.nealelab.is/blog/2017/9/13/heritability-201-types-of-heritability-and-how-we-estimate-it)). A longer discussion of the many limitations of this analysis is included in our more technical description of the project ([Heritability 501](http://www.nealelab.is/blog/2017/9/14/heritability-501-ldsr-based-h2-in-ukbb-for-the-technically-minded)).

This is a very rough initial analysis and should be treated accordingly. All estimates are for (a) the heritability from common genetic variants (b) of automatically cleaned and transformed phenotypes (c) in a non-random population sample of the UK (d) estimated with a fairly simple statistical model. We expect the results will mostly be useful for generating hypotheses for follow-up (e.g. identifying trends in the estimates for categories of phenotypes, choosing phenotypes with robust statistical evidence for heritability), rather than being a definitive statement about the “true” heritability of a individual phenotype. 

<p></p>

#### **Are genetic correlation results available for these phenotypes?**

Yes! You can browse genetic correlations between the significantly heritable phenotypes at [https://ukbb-rg.hail.is/](https://ukbb-rg.hail.is/). You can also download the LDSC-formatted sumstats files for your own genetic correlation analyses [here](). Round 1 results are also available for genetic correlation analyses through [LD Hub](http://ldsc.broadinstitute.org/).

</div>

<br>

<div class="well">

### Credits

* **Lead:** Raymond Walters
* **LDSR analysis support:** Nikolas Baya, Katherine Tashman, Danfeng Chen, Liam Abbott
* **Phenotype analysis support**: Caitlin Carey, Duncan Palmer
* **Supervision**: Benjamin Neale
* **UK Biobank GWAS Core Team:** Liam Abbott, Sam Bryant, Claire Churchhouse, Andrea Ganna, Daniel Howrigan, Duncan Palmer, Ben Neale, Raymond Walters, Caitlin Carey, The Hail team, Benjamin Neale

See [Credits](credits.html) for complete acknowledgements.

</div>
